src/odemis/gui/comp/overlay/shapes.py

```diff
@@ -180,39 +180,37 @@ def on_leave(self, evt):
         else:
             WorldOverlay.on_leave(self, evt)
 
-    def _activate_shapes(self, flag=True):
-        """Activate or de-activate the shapes."""
+    def _deselect_shapes(self):
+        """Deselect shapes except the selected shape."""
         for shape in self._shapes:
-            shape.active.value = flag
+            if shape.selected.value and shape != self._selected_shape:
+                shape.selected.value = False
 
     def _on_tool(self, selected_tool):
         """Update the overlay when it's active and tools change."""
-        if selected_tool == self.tool:
-            self.active.value = True
-            self._activate_shapes(True)
-        else:
-            self.active.value = False
-            self._activate_shapes(False)
-            self.cnvs.reset_default_cursor()
+        self.active.value = selected_tool == self.tool
 
-    def _get_shape(self, evt):
+    def _get_shape(self, v_pos: Tuple[float, float]):
         """
-        Find a shape corresponding to the given on_left_down event position.
+        Find a shape corresponding to the given view position.
         :return: the most appropriate shape.
             If no shape is found, it returns None.
         """
         if self._shapes:
-            pos = self.cnvs.view_to_phys(evt.Position, self.cnvs.get_half_buffer_size())
```

Comment:
Please indicate that this method is optimized for speed, and why (it's called for every mouse move?). Also indicate whether it's optimized for the "best case" (finding a shape) or "worst case" (no shape is below the given position.